kpet/data.py

```diff
@@ -423,6 +426,33 @@ def validate_host_type_regex(self):
                                                host_type_regex.pattern,
                                                host_types))
 
+    def validate_suite_origins(self):
+        """
+        Check that suite origins are valid.
+        Raises:
+            schema.Invalid when finding an invalid origin.
+        """
+        for suite in self.suites:
+            if self.origins is None:
```

Comment:
Shouldn't this fail also when `self.origins` is an empty dictionary? In that case, you can use `if not self.origins`.

Reply:
An empty dictionary is different from not having any specified at all. The former means no origin would be valid even though it could be soecified, and the latter means I cannot specify any. The distinction could be useful for e.g. automatically generating a kpet db and detecting when it failed generating any valid origins. I generally find it's best to avoid conflating cases, when we can properly support and distinguish them. It generally produces cleaner, simpler code.